src/core/src/storage/mod.rs

```diff
@@ -296,12 +297,19 @@ impl Storage for FSStorage {
     }
 
     fn load_sig(&self, path: &str) -> Result<SigStore> {
-        let raw = self.load(path)?;
-        let sig = Signature::from_reader(&mut &raw[..])?
-            // TODO: select the right sig?
-            .swap_remove(0);
+        let sig = match self.load(path) {
+            Ok(raw) => {
+                let sig = Signature::from_reader(&mut &raw[..])?
+                    .swap_remove(0)
+                    .into();
+                sig
+            },
+            Err(_) => {
+                todo!("cannot load from path '{path}'")
+            }
+        };
```

Comment:
I don't think this is an improvement, because the error is gone (replaced with a `todo!()`), and the `?` from `.load()` should already be returning the path that could not be loaded

Reply:
yes, it was the changes in the calling function that were important. reverted!